src/resolve_links.cpp

```diff
@@ -130,6 +143,41 @@ void resolve_links::match(std::shared_ptr<const torrent_info> const& ti
 	}
 
 }
+void resolve_links::match_v2(std::shared_ptr<const torrent_info> const& ti
+	, std::string const& save_path)
+{
+	file_storage const& fs = ti->files();
+	m_file_sizes.reserve(aux::numeric_cast<std::size_t>(fs.num_files()));
```

Comment:
The reason I ask is that in [torrent.cpp#L1850](https://github.com/arvidn/libtorrent/blob/RC_2_0/src/torrent.cpp#L1850) the resolver object is kept around as it iterates over similar torrents.  If someone puts a V1 torrent as similar in a V2 torrent it will just spin. They should never match though, right?  If so maybe I need to put a check in there so that only V1 and V2 torrents are to be considered as similar.  I apologize, this is a bit out of my depth so forgive me if these are tedious questions ;)

Reply:
> and they both have a file with matching file-root-hash, it's not really possible for those files to have different sizes. 

Actually, vanilla merkle trees are to preimage attacks if you don't specify the length or prefix data before hashing. So only the root hash + length uniquely identifies a file.

https://en.wikipedia.org/wiki/Merkle_tree#Second_preimage_attack

Reply:
I think there's a missing "are *subject* to".

Reply:
@the8472 I changed it to use a tuple (more or less) of the size and sha256, is that what you had in mind?

@arvidn I changed it so that the code was more in line with how the v1 sizes were tracked.  This allows it to behave the same way with respect to not bothering with files that have already been matched.  Probably not a big issue for low file count torrents, but at least in my mind, it looks marginally more consistent.

Reply:
> is that what you had in mind?

Yes.